src/veins/modules/mobility/traci/TraCIMobility.cc

```diff
@@ -296,6 +304,32 @@ double TraCIMobility::calculateCO2emission(double v, double a) const
     double zeta = 0.241;
     double alpha1 = 0.973;
 
+    // Eq 9 - please note: gamma was dropped in comparison to Eq 6
+    if (P_tract <= 0) return alpha1;
+    return alpha + beta * v * 3.6 + delta * v * v * v * (3.6 * 3.6 * 3.6) + zeta * a * v;
+}
+
+double TraCIMobility::calculateFuelConsumption(double v, double a) const
+{
+    // Calculate fuel consumption parameters according to:
+    // Cappiello, A. and Chabini, I. and Nam, E.K. and Lue, A. and Abou Zeid, M., "A statistical model of vehicle emissions and fuel consumption," IEEE 5th International Conference on Intelligent Transportation Systems (IEEE ITSC), pp. 801-809, 2002
+
+    double A = 1000 * 0.1326; // W/m/s
+    double B = 1000 * 2.7384e-03; // W/(m/s)^2
+    double C = 1000 * 1.0843e-03; // W/(m/s)^3
+    double M = 1325.0; // kg
+
+    // power in W
+    double P_tract = A * v + B * v * v + C * v * v * v + M * a * v; // for sloped roads: +M*g*sin_theta*v
+
+    // Table 1: calibrated parameters for the engine-out emissions module
+    double alpha = 0.365;
+    double beta = 0.00114;
+    double delta = 9.65e-07;
+    double zeta = 0.0943;
+    double alpha1 = 0.299;
```

Comment:
Are these constants? If so they should be qualified as `constexpr`. Also, if they are conceptually the same as used in `TraCIMobility::calculateCO2Emmision()`, they should rather be defined in a single location, e.g., an anonymous `namespace`.